components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/internal/SecretManagerComponent.java

```diff
@@ -0,0 +1,142 @@
+import org.wso2.carbon.identity.secret.mgt.core.dao.SecretDAO;
+import org.wso2.carbon.identity.secret.mgt.core.dao.impl.CachedBackedSecretDAO;
+import org.wso2.carbon.identity.secret.mgt.core.dao.impl.SecretDAOImpl;
+import org.wso2.carbon.identity.secret.mgt.core.model.SecretManagerConfigurationHolder;
+import org.wso2.carbon.user.core.service.RealmService;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * OSGi declarative services component which handles registration and un-registration of configuration management
+ * service.
+ */
+@Component(
+        name = "carbon.secret.mgt.component",
+        immediate = true
+)
+public class SecretManagerComponent {
+
+    private static final Log log = LogFactory.getLog(SecretManagerComponent.class);
+    private List<SecretDAO> secretDAOS = new ArrayList<>();
+
+    /**
+     * Register SecretManager as an OSGI service.
+     *
+     * @param componentContext OSGI service component context.
+     */
+    @Activate
+    protected void activate(ComponentContext componentContext) {
+
+        try {
+            BundleContext bundleContext = componentContext.getBundleContext();
+
+            SecretDAO secretDAO = new SecretDAOImpl();
+            bundleContext.registerService(SecretDAO.class.getName(), secretDAO, null);
+            bundleContext.registerService(SecretDAO.class.getName(),
+                    new CachedBackedSecretDAO(secretDAO), null);
+            SecretManagerConfigurationHolder secretManagerConfigurationHolder =
+                    new SecretManagerConfigurationHolder();
+            secretManagerConfigurationHolder.setSecretDAOS(secretDAOS);
+
+            bundleContext.registerService(SecretManager.class.getName(),
+                    new SecretManagerImpl(secretManagerConfigurationHolder), null);
+            bundleContext.registerService(SecretResolveManager.class.getName(),
+                    new SecretResolveManagerImpl(secretManagerConfigurationHolder), null);
+            SecretManagerComponentDataHolder.getInstance().setSecretManagementEnabled
+                    (isSecretManagementEnabled());
+        } catch (Throwable e) {
+            log.error("Error while activating SecretManagerComponent.", e);
+        }
+    }
+
+    @Reference(
+            name = "secret.dao",
+            service = org.wso2.carbon.identity.secret.mgt.core.dao.SecretDAO.class,
+            cardinality = ReferenceCardinality.MULTIPLE,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetSecret"
+    )
+    protected void setSecret(SecretDAO secretDAO) {
+
+        if (secretDAO != null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Secret DAO is registered in SecretManager service.");
+            }
+
+            this.secretDAOS.add(secretDAO);
+            this.secretDAOS.sort(Comparator.comparingInt(SecretDAO::getPriority));
```

Comment:
Other than maintaining all the secretDAOS, Can we keep only the Highest Priority DAO